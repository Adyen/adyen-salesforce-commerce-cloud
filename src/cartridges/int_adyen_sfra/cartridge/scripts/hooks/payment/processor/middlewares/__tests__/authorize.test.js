/* eslint-disable global-require */
let authorize;
let currentBasket;

beforeEach(() => {
  authorize = require('../authorize');
  jest.clearAllMocks();
  currentBasket = require('dw/order/BasketMgr').getCurrentBasket();
});

afterEach(() => {
  jest.resetModules();
});

describe('Authorize', () => {
  it('should return when create payment request fails', () => {
    const {
      createPaymentRequest,
    } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      error: {},
    }));
    const authorizeResult = authorize(
      '15',
      currentBasket.toArray()[0],
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });

  it('should authorize 3DS payments', () => {
    const {
      createPaymentRequest,
    } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      resultCode: 'RedirectShopper',
      redirectObject: {
        url: 'mockedUrl',
        data: {
          MD: 'mockedMD',
        },
      },
    }));
    const paymentInstrument = currentBasket.toArray()[0];
    const authorizeResult = authorize(
      '15',
      paymentInstrument,
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });

  it('should authorize 3DS2 payments', () => {
    const {
      createPaymentRequest,
    } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      threeDS2: 'mockedthreeDS2',
      resultCode: 'mockedresultCode',
      fullResponse: {action: 'mockedAction'},
    }));
    const authorizeResult = authorize(
      '15',
      currentBasket.toArray()[0],
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });

  it('should authorize redirectShopper payments', () => {
    const {
      createPaymentRequest,
    } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      resultCode: 'RedirectShopper',
      redirectObject: {
        url: 'mockedURL',
      },
      paymentData: 'mockedpaymentData',
    }));
    const paymentInstrument = currentBasket.toArray()[0];
    const authorizeResult = authorize(
      '15',
      paymentInstrument,
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });

  it('should handle the create payment request decision accept', () => {
    const {
      createPaymentRequest,
    } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      decision: 'ACCEPT',
    }));
    const authorizeResult = authorize(
      '15',
      currentBasket.toArray()[0],
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });

  it('should handle create payment request decisions other than accept', () => {
    const {
      createPaymentRequest,
    } = require('*/cartridge/scripts/adyenCheckout');
    createPaymentRequest.mockImplementation(() => ({
      decision: "DON'T ACCEPT",
    }));
    const authorizeResult = authorize(
      '15',
      currentBasket.toArray()[0],
      'mockedPaymentProcessor',
    );
    expect(authorizeResult).toMatchSnapshot();
  });
});
