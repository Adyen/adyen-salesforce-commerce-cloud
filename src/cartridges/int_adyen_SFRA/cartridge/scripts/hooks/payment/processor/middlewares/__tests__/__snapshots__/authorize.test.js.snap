// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorize should authorize 3DS payments 1`] = `
{
  "error": true,
}
`;

exports[`Authorize should authorize 3DS2 payments 1`] = `
{
  "error": true,
}
`;

exports[`Authorize should authorize redirectShopper payments 1`] = `
{
  "error": true,
}
`;

exports[`Authorize should handle create payment request decisions other than accept 1`] = `
{
  "error": true,
}
`;

exports[`Authorize should handle the create payment request decision accept 1`] = `
{
  "error": true,
}
`;

exports[`Authorize should return when create payment request fails 1`] = `
{
  "authorized": false,
  "error": true,
  "fieldErrors": [],
  "serverErrors": [
    "mocked_error.payment.processor.not.supported",
  ],
}
`;
